dkms_prebuild.sh

```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env sh
+
+test -x configure || ./autogen.sh
```

Comment:
In that case we can simply enhance `./autogen.sh` with something similar:
```diff
diff --git a/autogen.sh b/autogen.sh
index 2b47624..5f729c4 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,3 +1,17 @@
 #!/bin/sh

+if [ "$#" -gt 0 ]; then
+    if [ "$1" != --optional ]; then
+        echo "Usage: $0 [--optional]"
+        echo
+        echo "Options:"
+        echo "  --optional    Don't regenerate if already done"
+        exit 1
+    fi
+
+    if [ -x ./configure ]; then
+        exit 0
+    fi
+fi
+
 autoreconf -ivf || exit 1

```

Reply:
That is a cool thing to add to `autogen.sh`. However, it is still necessary to have `dkms_prebuild.sh` because `PRE_BUILD` only supports paths to a script, no arguments. But, with that addition, it would just become `./autogen.sh --optional` in `dkms_prebuild.sh`. This overall increases the complexity some. Thoughts on if that is worth it?

Reply:
then maybe in that case, using your `dkms_prebuild.sh` with `test -x configure || ./autogen.sh` is enough